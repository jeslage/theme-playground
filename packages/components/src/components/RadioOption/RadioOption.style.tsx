import styled from 'styled-components';

const StyledRadioOption = styled.label`
  position: relative;
  cursor: pointer;
  font-size: 12px;

  input {
    position: absolute !important;
    height: 1px;
    width: 1px;
    overflow: hidden;
    visibility: hidden;
    clip: rect(1px 1px 1px 1px); /* IE6, IE7 */
    clip: rect(1px, 1px, 1px, 1px);
    white-space: nowrap; /* added line */
  }

  div {
    display: flex;
    align-items: center;
    padding: 0.5rem 0.75rem;
    margin: 0 0 0 10px;
    border-radius: 3rem;
    opacity: 0.5;
    color: ${({ theme }) => theme.text};
    border: 1px solid ${({ theme }) => theme.border};
    transition: all 0.2s ease-in-out;

    svg {
      width: 20px;
      height: auto;
      fill: ${({ theme }) => theme.text};
    }
  }

  div:hover {
    opacity: 1;
  }

  input:checked + div {
    opacity: 1;
    background: ${({ theme }) => theme.text};
    color: ${({ theme }) => theme.backgroundDark};
  }
`;

export default StyledRadioOption;
