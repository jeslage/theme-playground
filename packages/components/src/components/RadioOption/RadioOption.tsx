import * as React from 'react';

import StyledRadioOption from './RadioOption.style';

export interface RadioOptionProps {
  onChange?: (val: string) => void;
  name?: string;
  value: string;
  label?: string;
  icon?: JSX.Element;
  isChecked?: boolean;
}

const RadioGroup: React.FC<RadioOptionProps> = ({
  label,
  name,
  value,
  onChange,
  isChecked,
  icon
}) => {
  return (
    <StyledRadioOption htmlFor={value}>
      <input
        type="radio"
        id={value}
        name={name}
        value={value}
        checked={isChecked}
        onChange={({ target }) => {
          if (onChange) {
            onChange((target as HTMLInputElement).value);
          }
        }}
      />
      <div>{icon || label || value}</div>
    </StyledRadioOption>
  );
};

export default RadioGroup;
