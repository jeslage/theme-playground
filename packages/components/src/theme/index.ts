export const darkTheme = {
  base: 'dark',
  backgroundDark: '#2f2f2f',
  backgroundLight: '#333',
  border: '#999999',
  text: '#FFFFFF'
};

export const lightTheme = {
  base: 'light',
  backgroundDark: '#f6f9fc',
  backgroundLight: '#fff',
  border: '#666666',
  text: '#333333'
};

export const getUiTheme = theme => {
  if (theme === undefined) return darkTheme;

  if (typeof theme === 'string') {
    if (theme === 'dark') {
      return darkTheme;
    } else if (theme === 'light') {
      return lightTheme;
    }

    throw Error(
      `Theme "${theme}" is not available. Please use "light" or "dark", or provide a custom theme object.`
    );
  }

  if (theme.base && theme.base === 'light') {
    return {
      ...lightTheme,
      ...theme
    };
  }

  return {
    ...darkTheme,
    ...theme
  };
};
