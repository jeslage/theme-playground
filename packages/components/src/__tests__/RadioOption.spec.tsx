import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';

import RadioOption from '../components/RadioOption/RadioOption';
import StyledRadioOption from '../components/RadioOption/RadioOption.style';
import { darkTheme } from '../theme';

describe('RadioOption', () => {
  it('RadioOption should render correctly', () => {
    const component = shallow(<RadioOption label="primary" value="primary" />);

    expect(component).toMatchSnapshot();
  });

  it('RadioOption without label should render correctly', () => {
    const component = shallow(<RadioOption value="primary" />);

    expect(component).toMatchSnapshot();
  });

  it('StyledRadioOption should render correctly', () => {
    const component = shallow(
      <StyledRadioOption theme={{ colors: darkTheme }} />
    );

    expect(component).toMatchSnapshot();
  });
});
