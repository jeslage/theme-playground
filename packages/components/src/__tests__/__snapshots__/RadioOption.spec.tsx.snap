// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioOption RadioOption should render correctly 1`] = `
<styled.label
  htmlFor="primary"
>
  <input
    id="primary"
    onChange={[Function]}
    type="radio"
    value="primary"
  />
  <div>
    primary
  </div>
</styled.label>
`;

exports[`RadioOption RadioOption without label should render correctly 1`] = `
<styled.label
  htmlFor="primary"
>
  <input
    id="primary"
    onChange={[Function]}
    type="radio"
    value="primary"
  />
  <div>
    primary
  </div>
</styled.label>
`;

exports[`RadioOption StyledRadioOption should render correctly 1`] = `
<label
  className="sc-AxjAm fEmwmh"
/>
`;
